### DISCUSSION TITLE
Multiple possible literals

### DISCUSSION THREAD

**Discussion by hinogi** [colinhacks/zod#1379] [Q&A]
Is there a way to have multiple possible literals somehow?
this is what I tried, didn't seem to work.
```ts
z
  .object({
    __typename: z
      .literal('A')
      .or(z.literal('B'))
      .or(z.literal('C'))
  })
  .optional()
```

**Comment by ariesclark** [comment]
Something like this?
```ts
z.union([z.literal("A"), z.literal("B"), z.literal("C")])
```

**Reply by hinogi** [reply]
that is the other form of the `.or` syntax :D

**Reply by taylorkline** [reply]
ariesclark's solution works for me.

**Reply by taylorkline** [reply]
Never mind, the specific way that it "didn't seem to work" is that you get a failure at runtime:


> A discriminator value for key could not be extracted from all schema options
